test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/spot-instance-machine-pool.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```

Comment:
I don't see this template being used in any e2e tests. Should there also be a change in `e2e_conf.yaml` and then use the flavour in a test.

Or have i missed its usage?

Reply:
this is used in kustomization yaml itself https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/1e938c88ebe75470f11e0159fc24d8f8212da03b/test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/kustomization.yaml#L4
We have not created separate e2e test as CAPI MachinePool test expect only one flavor, so we have combined both in 1 test.

Reply:
Got you, thanks @Ankitasw.

Do we want to combine the tests long term?

Reply:
I knowingly kept these two as separate tests, as both are for different use cases.

Reply:
> Good point @sedefsavas , I checked earlier in logs only if machine pool is created, it showed created in logs, but looks like its not creating 2nd machine pool in aws console

@Ankitasw this is resolved now? 
CAPI MP test creates both the MPs?

Reply:
yes this is resolved.